packages/mui-base/src/InputUnstyled/InputUnstyled.test.tsx

```diff
@@ -30,4 +33,26 @@ describe('<InputUnstyled />', () => {
 
     expect(screen.getByRole('textbox')).to.have.tagName('textarea');
   });
+
+  describe('componentsProps', () => {
+    it('allows to pass in a callback function', () => {
+      const componentProps = (ownerState: InputUnstyledOwnerState) => ({
+        className: ownerState ? 'test-focused' : '',
```

Comment:
```suggestion
        className: ownerState.focused ? 'test-focused' : '',
```

Was this the intent?

Reply:
Yes, thanks for noticing!